build-combo.sh

```diff
@@ -0,0 +1,104 @@
+#! /bin/bash
+set -eo pipefail
+trap ctrl_c INT
+
+readonly YELLOW=$(tput setaf 3)
+readonly NC=$(tput sgr0)
+
+readonly BACKUP_FILE=".dazzle.yaml.orig"
+readonly ORIGINAL_FILE="dazzle.yaml"
+
+function save_original {
+    if [ ! -f ${BACKUP_FILE} ]; then
+        echo "${YELLOW}Creating a backup of ${ORIGINAL_FILE} as it does not exist yet...${NC}" && cp ${ORIGINAL_FILE} ${BACKUP_FILE}
+    fi
+}
+
+function restore_original() {
+	echo "${YELLOW}Restoring backup file ${BACKUP_FILE} to original file ${ORIGINAL_FILE}${NC}"
+	cp ${BACKUP_FILE} ${ORIGINAL_FILE}
+}
+
+function ctrl_c() {
+	echo "${YELLOW}** Trapped CTRL-C${NC}"
+	restore_original
+}
```

Comment:
Can we extract line 2 to 25 into a separate `common.sh` and use it here and in `build-chunk.sh`